WordPress/gutenberg
# Spacer: Complete TypeScript migration of component

## What?
Completes the migration of `Spacer` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored test to use TypeScript
- Updated docs
- Sorted props in readme and types alphabetically 
- Aligned readme and types documentation

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Spacer` continues to function as expected
